scipy/spatial/qhull.pyi

```diff
@@ -2,41 +2,52 @@
 Static type checking stub file for scipy/spatial/qhull.pyx
 '''
 
-from __future__ import annotations
-from typing import TYPE_CHECKING, List, Tuple, Any
-
-if TYPE_CHECKING:
-    from numpy.typing import ArrayLike
-else:
-    ArrayLike = Any
+from typing import List, Tuple, Any, Dict
 
 import numpy as np
+from numpy.typing import ArrayLike
+from typing_extensions import final
 
+@final
```

Comment:
This is useful - I can remember quite a few problems that could have been prevented by judicious use of `@final`.